tools/end_to_end_test_llama.py

```diff
@@ -0,0 +1,324 @@
+                ["Michael, an American Jew, called Jews"],
+                ["Blackhawks\n The 2015 Hilltoppers"],
+                ["Data sources you can use to make a comparison:"]
+                ]
+        bad_words_list = np.array([
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""]], dtype=object)
+        stop_words_list = np.array([
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""],
+            [""]], dtype=object)
+        input0_data = np.array(input0).astype(object)
+        output0_len = np.ones_like(input0).astype(np.uint32) * OUTPUT_LEN
+        runtime_top_k = (FLAGS.topk * np.ones([input0_data.shape[0], 1])).astype(np.uint32)
+        runtime_top_p = FLAGS.topp * np.ones([input0_data.shape[0], 1]).astype(np.float32)
+        beam_search_diversity_rate = 0.0 * np.ones([input0_data.shape[0], 1]).astype(np.float32)
+        temperature = 1.0 * np.ones([input0_data.shape[0], 1]).astype(np.float32)
+        len_penalty = 1.0 * np.ones([input0_data.shape[0], 1]).astype(np.float32)
+        repetition_penalty = 1.0 * np.ones([input0_data.shape[0], 1]).astype(np.float32)
+        random_seed = 0 * np.ones([input0_data.shape[0], 1]).astype(np.uint64)
+        is_return_log_probs = True * np.ones([input0_data.shape[0], 1]).astype(bool)
+        beam_width = (FLAGS.beam_width * np.ones([input0_data.shape[0], 1])).astype(np.uint32)
+        start_ids = start_id * np.ones([input0_data.shape[0], 1]).astype(np.uint32)
+        end_ids = end_id * np.ones([input0_data.shape[0], 1]).astype(np.uint32)
+        inputs = [
+            prepare_tensor("INPUT_0", input0_data, FLAGS.protocol),
+            prepare_tensor("INPUT_1", output0_len, FLAGS.protocol),
+            prepare_tensor("INPUT_2", bad_words_list, FLAGS.protocol),
+            prepare_tensor("INPUT_3", stop_words_list, FLAGS.protocol),
+            prepare_tensor("runtime_top_k", runtime_top_k, FLAGS.protocol),
+            prepare_tensor("runtime_top_p", runtime_top_p, FLAGS.protocol),
+            prepare_tensor("beam_search_diversity_rate", beam_search_diversity_rate, FLAGS.protocol),
+            prepare_tensor("temperature", temperature, FLAGS.protocol),
+            prepare_tensor("len_penalty", len_penalty, FLAGS.protocol),
+            prepare_tensor("repetition_penalty", repetition_penalty, FLAGS.protocol),
+            prepare_tensor("random_seed", random_seed, FLAGS.protocol),
+            prepare_tensor("is_return_log_probs", is_return_log_probs, FLAGS.protocol),
+            prepare_tensor("beam_width", beam_width, FLAGS.protocol),
+            prepare_tensor("start_id", start_ids, FLAGS.protocol),
+            prepare_tensor("end_id", end_ids, FLAGS.protocol),
+        ]
+
+        try:
+            result = client.infer(model_name, inputs)
```

Comment:
I see the llama model is decoupled , so shouldn't the call be async_stream_infer  instead of  infer?

Reply:
I correct it to start_stream(call_back（）) and async_stream_infer() and use old input（HttpInferInput）， but got an error "TypeError: Not a cmessage" from tritonclient/grpc/_utils.py